### PR TITLE
CIP-26: Get validator BLS key precompile

### PR DESCRIPTION
**Pull request by mrsmkl**
### Description

https://github.com/celo-org/celo-proposals/pull/84

Depends on CIP22 PR (https://github.com/celo-org/celo-blockchain/pull/1231) because it needs the latest celo-bls-go.

### Other changes

To speed it up, added the uncompressed BLS key to the snapshots. (Many possibilities how this should be done)

### Tested

Added unit tests.

### Backwards compatibility

Will require hardfork.

### CODE REVIEW

File: params/protocol_params.go
Comment (kobigurk) [inline params/protocol_params.go:140]:
```suggestion
	GetValidatorBLSGas          uint64 = 1000   // Cost of reading a validator's BLS public key.
```

File: params/config.go
Comment (kobigurk) [inline params/config.go:142]: We need to check if this is desired. I think it is, but let's make sure. @mcortesi?

**Review by kobigurk — CHANGES_REQUESTED** [review]
LGTM! Please merge the latest state from the branch you've branched from, and let's mark it as "do not merge" until the other one is merged.

File: params/config.go
Comment (mcortesi) [reply params/config.go:142]: I would wait for now

**Review by mcortesi — COMMENTED** [review]
I would chat with @prestwich, he has a PR with scaffold for Donut precompiles which has does part of this

**Comment by prestwich** [comment]
This one: 
https://github.com/celo-org/celo-blockchain/pull/1276

It'd be good to merge a single PR with all the precompile address scaffolding, and then we can more easily merge precompiles as they're ready

**Comment by mcortesi** [comment]
> This one:
> #1276
> 
> It'd be good to merge a single PR with all the precompile address scaffolding, and then we can more easily merge precompiles as they're ready

thanks for the link! I did copy the link with the intention of pasting it :(  ....

File: core/vm/contracts.go
Comment (prestwich) [inline core/vm/contracts.go:78]: This is now defined in `master`

File: core/vm/contracts.go
Comment (prestwich) [inline core/vm/contracts.go:158]: This block is now defined in `master`

**Review by prestwich — CHANGES_REQUESTED** [review]
With #1276, the precompile address and map definitions for Donut have been merged into `master`.   Can you please rebase onto latest `master`?

File: params/config.go
Comment (kobigurk) [reply params/config.go:142]: @mcortesi can you look again? maybe it already makes sense to activate donut in dev config

Comment (mcortesi) [reply params/config.go:142]: let's leave as it is

**Review by kobigurk — APPROVED** [review]

**Review by prestwich — APPROVED** [review]

**Pull request merged**
